test/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/EnterMovementReferenceNumberControllerSpec.scala

```diff
@@ -105,23 +109,29 @@ class EnterMovementReferenceNumberControllerSpec
 
       def performAction(): Future[Result] = controller.enterJourneyMrn(JourneyBindable.Single)(FakeRequest())
 
-      "show the title" in {
+      "show the title" in forAll(keys) { key =>
+        featureSwitch.EntryNumber.setFlag(key != enterNoLegacyMrnKey)
+
         val (session, _, _) = sessionWithClaimState(None, Individual)
+
         inSequence {
           mockAuthWithNoRetrievals()
           mockGetSession(session)
         }
-        val doc             = Jsoup.parse(contentAsString(performAction()))
 
-        doc.select("h1").text                                    should include(messageFromMessageKey("enter-movement-reference-number.title"))
+        val doc = Jsoup.parse(contentAsString(performAction()))
+
+        doc.select("h1").text                                    should include(messageFromMessageKey(s"$key.title"))
         doc.select("#enter-movement-reference-number").`val`() shouldBe ""
       }
     }
 
     "Change MRN page" must {
       def performAction(): Future[Result] = controller.changeJourneyMrn(JourneyBindable.Single)(FakeRequest())
 
-      "show the title and the MRN number" in {
+      "show the title and the MRN number" in forAll(keys) { key =>
+        featureSwitch.EntryNumber.setFlag(key != enterNoLegacyMrnKey)
+
         val mrn             = MRN("10ABCDEFGHIJKLMNO0")
```

Comment:
we can use the MRN generator here

Reply:
It is something I asked @caoilte to do back then. I should be addressed by merging master.

Reply:
yes, and you have merge conflicts now which need to be addressed